es/es-server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -89,7 +90,7 @@ public static Version fromId(int internalId) {
             case ES_V_6_1_4_ID:
                 return ES_V_6_1_4;
             case ES_V_7_0_0_ID:
-                return V_4_0_0;
+                return V_4_0_1;
```

Comment:
not sure whether i get the versions mapping right.

Reply:
Starting with 4.0.0 we should have a 1:1 mapping, so something like this:

```
--- a/es/es-server/src/main/java/org/elasticsearch/Version.java
+++ b/es/es-server/src/main/java/org/elasticsearch/Version.java
@@ -70,7 +70,10 @@ public class Version implements Comparable<Version>, ToXContentFragment {
     public static final int ES_V_7_0_0_ID = 7_00_00_99;
     public static final Version V_4_0_0 = new Version(ES_V_7_0_0_ID, ES_V_7_0_0_ID - INTERNAL_OFFSET, false, org.apache.lucene.util.Version.LUCENE_8_0_0);
 
-    public static final Version CURRENT = V_4_0_0;
+    public static final int ES_V_7_0_1_ID = 7_00_01_99;
+    public static final Version V_4_0_1 = new Version(ES_V_7_0_1_ID, ES_V_7_0_1_ID - INTERNAL_OFFSET, true, org.apache.lucene.util.Version.LUCENE_8_0_0);
+
+    public static final Version CURRENT = V_4_0_1;
 
 
     static {
@@ -90,6 +93,8 @@ public class Version implements Comparable<Version>, ToXContentFragment {
                 return ES_V_6_1_4;
             case ES_V_7_0_0_ID:
                 return V_4_0_0;
+            case ES_V_7_0_1_ID:
+                return V_4_0_1;
             case V_EMPTY_ID:
                 return V_EMPTY;
             default:
```

See the comment here as explanation:

https://github.com/crate/crate/blob/082a5efdc48ffde27225c06fbb2784605a9661d7/es/es-server/src/main/java/org/elasticsearch/Version.java#L53-L67